<launch>

  <!-- Arguments -->
  <arg name="config_path" default="$(find pose_graph)/config/gopro/gopro_uw.yaml" />

  <param name="/use_sim_time" value="true" />

  <!-- To un-compress image topics -->
  <node name="republish" type="republish" pkg="image_transport" args="compressed in:=/gopro/image_raw raw out:=/cam0/image_raw" />


  <!--Run pose_graph node-->
  <node name="pose_graph_node" pkg="pose_graph" type="pose_graph_node" output="screen">
    <param name="config_file" type="string" value="$(arg config_path)" />
  </node>


  <node name="okvis_node" pkg="okvis_ros" type="okvis_node">

    <param name="config_filename" value="$(find okvis_ros)/okvis/config/config_gopro_uw_radtan.yaml" />

    <remap from="/camera0" to="/cam0/image_raw" />
    <remap from="/imu" to="/gopro/imu" />

  </node>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find okvis_ros)/config/rviz_svin.rviz" />


</launch>